### PR TITLE
Add toSdkBindingDataMap() to SdkType

### PR DESCRIPTION
**Pull request by andresgomezfrr**
# TL;DR
Add a new method into the SdkType interface to allow to transform of an input/output class in a `Map<String, SdkBindingData<?>>`

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Add a new method into the SdkType interface to allow to transform an input/output class in a Map<String, SdkBindingData<?>>

This PR also adds their implementations ([JacksonSdkType](https://github.com/flyteorg/flytekit-java/blob/a9677418737988766afd81a0dcc6e53c673e8ffb/flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/JacksonSdkType.java), [SdkScalaType](https://github.com/flyteorg/flytekit-java/blob/a9677418737988766afd81a0dcc6e53c673e8ffb/flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkScalaType.scala))

## Tracking Issue

* https://github.com/flyteorg/flyte/issues/3258

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit-jackson/src/test/java/org/flyte/flytekit/jackson/JacksonSdkTypeTest.java [hunk 1]
```diff
@@ -283,6 +286,76 @@ void testToLiteralMap() {
...
+    assertEquals(expected, sdkBindingDataMap);
+  }
+
+  @Test
+  public void testToSdkBindingDataMapJsonProperties() {
+
+    JsonPropertyClassInput input =
+        new JsonPropertyClassInput(
+            SdkBindingData.ofString("test"), SdkBindingData.ofString("name"));
+    System.out.println(JacksonSdkType.of(JsonPropertyClassInput.class).getVariableMap());
```
Comment (narape) [inline flytekit-jackson/src/test/java/org/flyte/flytekit/jackson/JacksonSdkTypeTest.java:334]:
```suggestion
```
You forgot this

File: flytekit-jackson/src/test/java/org/flyte/flytekit/jackson/JacksonSdkTypeTest.java [hunk 2]
```diff
@@ -283,6 +286,76 @@ void testToLiteralMap() {
...
+
+    Map<String, SdkBindingData<?>> sdkBindingDataMap =
+        JacksonSdkType.of(JsonPropertyClassInput.class).toSdkBindingMap(input);
+
+    Map<String, SdkBindingData<?>> expected = new HashMap<>();
+    expected.put("test", input.test);
+    expected.put("name", input.otherTest);
```
Comment (narape) [inline flytekit-jackson/src/test/java/org/flyte/flytekit/jackson/JacksonSdkTypeTest.java:341]:
```suggestion
    var expected = Map.of("test", input.test, "name", input.otherTest);
```
The Sdk was written when we were supporting Java8 but now the minimum in Java11, so lets take the opportunity to modernize code when we can

**Review by narape — COMMENTED** [review]
Well done, minor comments

File: flytekit-jackson/src/test/java/org/flyte/flytekit/jackson/JacksonSdkTypeTest.java (hunk 1 above)
Comment (andresgomezfrr) [reply flytekit-jackson/src/test/java/org/flyte/flytekit/jackson/JacksonSdkTypeTest.java:334]: 🙃

**Review by narape — APPROVED** [review]
